pkg/resources/warehouse.go

```diff
@@ -2,191 +2,251 @@ package resources
 
 import (
 	"context"
-
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/logging"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
-	snowflakevalidation "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/validation"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+// TODO [SNOW-1348102 - next PR]: extract three-value logic
+// TODO [SNOW-1348102 - next PR]: handle conditional suspension for some updates (additional optional field)
 var warehouseSchema = map[string]*schema.Schema{
 	"name": {
 		Type:        schema.TypeString,
 		Required:    true,
 		Description: "Identifier for the virtual warehouse; must be unique for your account.",
 	},
-	"comment": {
-		Type:     schema.TypeString,
-		Optional: true,
-		Default:  "",
-	},
-	"warehouse_size": {
+	"warehouse_type": {
 		Type:         schema.TypeString,
 		Optional:     true,
-		Computed:     true,
-		ValidateFunc: snowflakevalidation.ValidateWarehouseSize,
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-			oldSize, err := sdk.ToWarehouseSize(old)
-			if err != nil {
-				return false
-			}
-			newSize, err := sdk.ToWarehouseSize(new)
-			if err != nil {
-				return false
-			}
-			return oldSize == newSize
-		},
-		Description: "Specifies the size of the virtual warehouse. Larger warehouse sizes 5X-Large and 6X-Large are currently in preview and only available on Amazon Web Services (AWS).",
+		ValidateFunc: validation.StringInSlice(sdk.ValidWarehouseTypesString, true),
```

Comment:
Yep, forgot to do it.

Added diff suppression and changed validations.

it would be nice to do something more. It's now nice that we use the same normalized function for both validation and suppression, but still listed values come from additional lists. I will add this to "thoughts".